### PR TITLE
[Metrics] Supply extra metrics to monitor the flows under `prefer-leader` mode.

### PR DESCRIPTION
**Pull request by LykxSassinator**
## Description
This pr is used to supply extra necessary metrics for monitoring the dataflows under `prefer-leader` mode, providing an intuitive understanding if the destination of dataflows meet the expectation.

In this pr, the report of dataflows' statistics is batched and scheduled to call `prometheus` API for better accessing performance.

### CODE REVIEW

**Comment by disksing** [comment]
@LykxSassinator is there anyone you prefer to review the PR?

**Comment by LykxSassinator** [comment]
> @LykxSassinator is there anyone you prefer to review the PR?

@tonyxuqqi PTAL, thx

File: internal/locate/region_cache.go [hunk 1]
```diff
@@ -2758,6 +2763,42 @@ func (c *RegionCache) checkAndUpdateStoreSlowScores() {
...
+// asyncReportStoreReplicaFlows reports the statistics on the related replicaFlowsType.
+func (c *RegionCache) asyncReportStoreReplicaFlows(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.ctx.Done():
+			return
+		case <-ticker.C:
+			c.storeMu.RLock()
```
Comment (tonyxuqqi) [inline internal/locate/region_cache.go:2790]: Is there any critical code path require the Lock()?

**Review by tonyxuqqi — APPROVED** [review]

File: internal/locate/region_cache.go (hunk 1 above)
Comment (LykxSassinator) [reply internal/locate/region_cache.go:2790]: `storeMu` is used to assure the tick get a full snapshot of `stores`, avoiding to access equivocal or incomplete stores, which might exists during the period of updating stores by other threads.

**Comment by LykxSassinator** [comment]
ping @disksing , PTAL, thx.

**Pull request merged**